arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,37 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+const int TRIGGER_PIN           = 6; 
+const int ECHO_PIN              = 7; 
+const unsigned int MAX_DISTANCE = 200;
+const int FRONT_PIN = 0;
+
+ArduinoRuntime arduinoRuntime;
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+GP2Y0A21 frontIR(arduinoRuntime, FRONT_PIN); 
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+SimpleCar car(control);
+
+void setup()
+{
+    Serial.begin(9600);
+    car.setSpeed(30);
 }
 
-void loop() {
-  // put your main code here, to run repeatedly:
+void loop()
+{
+  if(front.getDistance() > 170)
+  {
+    car.setAngle(90);
+    car.setSpeed(15);
+    delay(3000);
+    car.setAngle(0);
+    car.setSpeed(30);
+  }
+  if (frontIR.getDistance() > 30)
+  {
+    car.setSpeed(0);
+  }
```

Comment:
Are you sure this is the correct logic? Can you explain it to me?

Reply:
What it does is that it starts turning the car when the distance from an object is 170 and at the same time we slow it down. After 3 seconds we return the angle and the speed to the start values. 
If the car gets too close to an object (30), the car speed is set to 0 to not crash the car.

Reply:
What happens if the distance is 20?! Crash on it?

Reply:
True, we have updated the logic a bit.